include/networkit/components/StronglyConnectedComponents.hpp

```diff
@@ -1,70 +1,143 @@
+     */
+    TLX_DEPRECATED(StronglyConnectedComponents(const Graph &G, bool iterativeAlgo));
+
+    /**
+     * Runs the algorithm.
      */
-    void run();
+    void run() override;
 
     /**
      * This method determines the connected components for the graph g
      * (iterative implementation).
+     *
+     * This method is deprecated, run() already uses the iterative implementation.
      */
-    void runIteratively();
+    void TLX_DEPRECATED(runIteratively());
 
     /**
      * This method determines the connected components for the graph g
      * (recursive implementation).
+     *
+     * This method is deprecated, use run().
+     */
+    void TLX_DEPRECATED(runRecursively());
+
+    /**
+     * Return the number of connected components.
+     *
+     * @return count The number of components of the graph.
+     */
+    count numberOfComponents() const {
+        assureFinished();
+        return componentIndex;
+    }
+
+    /**
+     * Returns the component of the input node.
+     *
+     * @param[in] u The input node.
+     * @return count The index of the component of the input node.
      */
-    void runRecursively();
+    index componentOfNode(node u) const {
+        assureFinished();
+        assert(G.hasNode(u));
+        return component[u];
+    }
 
     /**
-     * This method returns the number of connected components.
+     * Return a Partition object that represents the components.
+     *
+     * @return Partition Partitioning of the strongly connected components.
      */
-    count numberOfComponents();
+    Partition getPartition() const {
+        assureFinished();
+        return Partition(component);
+    }
 
     /**
-     * This method returns the the component in which node query is situated.
+     * Return a map with the component indexes as keys, and their size as values.
      *
-     * @param[in]	query	the node whose component is asked for
+     * @return std::map<index, count> Map with components indexes as keys, and their size as values.
      */
-    count componentOfNode(node u);
+    std::map<node, count> getComponentSizes() const {
+        assureFinished();
+        std::map<node, count> map;
+        for (index i = 0; i < componentIndex; ++i) {
+            map[i] = 0;
+        }
 
+        G.forNodes([&](const node u) { ++map[component[u]]; });
+        return map;
+    }
 
     /**
-     * Return a Partition that represents the components
+     * Return a list of components.
+     *
+     * @return std::vector<std::vector<node>> List of components.
      */
-    Partition getPartition();
+    std::vector<std::vector<node>> getComponents() const {
+        assureFinished();
+        std::vector<std::vector<node>> result(componentIndex);
+
+        G.forNodes([&](const node u) { result[component[u]].push_back(u); });
 
+        return result;
+    }
 
 private:
-    const Graph& G;
+    const Graph &G;
     bool iterativeAlgo;
-    Partition component;
+    std::vector<index> component;
```

Comment:
Maybe not for this PR - but if changing the datatype is the preferred solution for `component` - I think we should refactor all other component-types (or refactor the header-inheritance since they are all quite similar).

Reply:
Probably, I find `std::vector` to be a simpler choice here, and it does not restrict the features of the class.